source/MRMesh/MRObjectLinesHolder.cpp

```diff
@@ -30,6 +30,11 @@ void ObjectLinesHolder::applyScale( float scaleFactor )
     setDirtyFlags( DIRTY_POSITION );
 }
 
+bool ObjectLinesHolder::valid() const
+{
+    return polyline_ && polyline_->topology.lastNotLoneEdge().valid();
```

Comment:
better check via `topology.numValidVerts()`, which is cached. And valid vertex means at least one valid edge from it.

Reply:
Fixed.